### DISCUSSION TITLE
Question about design decision to have no package registry

### DISCUSSION THREAD

**Discussion by jtheisen** [denoland/deno#9597] [General]
NPM has a public registry you publish your package to. So obviously if that one moves or is replaced by a local one it won't require all applications to have all their files touched in order to reflect a changed url.

How is this handled in Deno?

I apologize for the filing a ticket for this and I'm only asking out of curiosity, but I did some 15m of digging before I did.

The best I could find [was this](https://deno.land/manual@v1.7.5/linking_to_external_code#it-seems-unwieldy-to-import-urls-everywhere), but that's only a subset of the issue. Specificall later it states about a failing source:

> This, like the above, is a problem faced by any remote dependency system.

That's technically true, but then I only need to replace _one_ source (the central registry), not _one per dependency_.

While Deno could offer a feature to remap the urls for a project, I would then still struggle to see what the point was to begin with.

**Comment by lucacasonato** [comment]
https://deno.land/manual@v1.7.5/linking_to_external_code/import_maps#import-maps are the other half. For more info join us on discord.gg/deno.

**Comment by jtheisen** [comment]
I edited in the last two lines just as the comment came.

**Comment by nayeemrmn** [comment]
> > This, like the above, is a problem faced by any remote dependency system.
> 
> That's technically true, but then I only need to replace _one_ source (the central registry), not _one per dependency_.

That assumes centralised name resolution between registries. If a host goes down, it makes sense to revisit individual URLs which use it. If all of those URLs have the same structure and the package name mapping for the new registry is the same, it's still a really simple search and replace.
